contracts/tests/verifier.test.ts

```diff
@@ -367,50 +360,621 @@ describe('Verifier Contract', () => {
+
+      if (_createCommit) {
+        await verifier.commit(testModelContentId, testMerkleRoot);
+      }
+
+      return [testModelContentId, testMerkleRoot];
+    };
+
+    it('success', async () => {
+      const [testModelContentId, testMerkleRoot] = await setup();
+
+      const commitId = generateCommitId(testModelContentId, testMerkleRoot);
+
+      const originalCommit = await verifier.getCommit(commitId);
+
+      const tx = await verifier.updateChallenge(commitId);
+
+      await expect(tx)
+        .to.emit(verifier, 'ChallengeUpdated')
+        .withArgs(commitId, ownerAddress);
+
+      const updatedCommit = await verifier.getCommit(commitId);
+      expect(updatedCommit.challenge).to.not.equal(originalCommit.challenge);
+    });
+
+    it('failure: commit not found', async () => {
+      const [testModelContentId, testMerkleRoot] = await setup(false, false);
+
+      const commitId = generateCommitId(testModelContentId, testMerkleRoot);
+
+      await expect(verifier.updateChallenge(commitId)).to.be.revertedWith(
+        'commit not found',
+      );
+    });
+
+    it('failure: invalid prover', async () => {
+      const [testModelContentId, testMerkleRoot] = await setup();
+
+      const commitId = generateCommitId(testModelContentId, testMerkleRoot);
+
+      await expect(
+        verifier.connect(anotherAccount).updateChallenge(commitId),
+      ).to.be.revertedWith('invalid prover');
+    });
+  });
+
+  describe('getDifficulty', () => {
+    it('success', async () => {
+      const _difficulty = await verifier.getDifficulty();
+
+      expect(_difficulty).to.equal(difficulty);
+    });
+
+    it('failure: not contract owner', async () => {
+      await expect(
+        verifier.connect(anotherAccount).getDifficulty(),
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+  });
+
+  describe('updateDifficulty', () => {
+    const validDifficulty = 11;
+    const invalidDifficulty = 0;
+
+    it('success', async () => {
+      await verifier.updateDifficulty(validDifficulty);
+
+      const updatedDifficulty = await verifier.getDifficulty();
+      expect(updatedDifficulty).to.equal(validDifficulty);
+    });
+
+    it('failure: invalid difficulty', async () => {
+      await expect(
+        verifier.updateDifficulty(invalidDifficulty),
+      ).to.be.revertedWith('difficulty cannot be 0');
+    });
+
+    it('failure: not contract owner', async () => {
+      await expect(
+        verifier.connect(anotherAccount).updateDifficulty(validDifficulty),
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+  });
+
+  describe('reveal', () => {
+    const setup = async (
+      _numOfNode = 10,
+      _createModel = true,
+      _createCommit = true,
+    ): Promise<[string, string, StandardMerkleTree<string[]>]> => {
+      // set difficulty to 1 to make sure that challenge almost always matches
+      // at least one of the autogenerated leaves
+      await verifier.updateDifficulty(1);
```

Comment:
Set `difficulty` to `1` in tests (i.e. probability of autogenerated Merkle tree containing the leaves matched with the autogenerated challenge is circa 50%).